raft/src/main/java/org/apache/kafka/raft/QuorumState.java

```diff
@@ -369,6 +379,17 @@ public void transitionToLeader(long epochStartOffset) throws IOException {
         if (!candidateState.isVoteGranted())
             throw new IllegalStateException("Cannot become leader without majority votes granted");
 
+        // Note that the leader does not retain the high watermark that was known
+        // in the previous state. The purpose of this is to protect the monotonicity
```

Comment:
What is "this" in this sentence? epochStartOffset?

Reply:
It was intended to refer to the behavior of not retaining the high watermark from the previous sentence. I will attempt to clarify.